src/covid_model_seiir/ode_forecasting/ode_runner.py

```diff
@@ -1,10 +1,67 @@
+from typing import Union
 from dataclasses import dataclass, asdict
 
 import numpy as np
 import pandas as pd
 
 from odeopt.ode import RK4
-from odeopt.ode.system.nonlinearsys import BetaSEIIR
+from odeopt.ode import ODESys
+
+
+class CustomizedSEIIR(ODESys):
+    """Customized SEIIR ODE system."""
+
+    def __init__(self,
+                 alpha: float,
+                 sigma: float,
+                 gamma1: float,
+                 gamma2: float,
+                 N: Union[int, float],
+                 delta: float, *args):
+        """Constructor of CustomizedSEIIR.
+        """
+        self.alpha = alpha
+        self.sigma = sigma
+        self.gamma1 = gamma1
+        self.gamma2 = gamma2
+        self.N = N
+        self.delta = delta
+
+        # create parameter names
+        params = ['beta', 'theta']
+
+        # create component names
+        components = ['S', 'E', 'I1', 'I2', 'R']
+
+        super().__init__(self.system, params, components, *args)
+
+    def system(self, t: float, y: np.ndarray, p: np.ndarray) -> np.ndarray:
+        """ODE System.
+        """
+        beta = p[0]
+        theta = p[1]
+
+        s = y[0]
+        e = y[1]
+        i1 = y[2]
+        i2 = y[3]
+        r = y[4]
+
+        theta_plus = max(theta, 0.)
+        theta_minus = min(theta, 0.)
+        theta_tilde = int(theta_plus != theta_minus)
+        theta_minus_alt = (self.gamma1 - self.delta) * i1 - self.sigma * e - theta_plus
```

Comment:
Might be a silly point (please ignore it if it is not helpful), could we assume that `self.delta > self.gamma1`, otherwise, when `theta < 0`, the `I1` might be decreasing slower than it original will be (only happens at down slope).
And we will know that `theta_minus_alt < 0` and we don't need to worry about `effective_theta_minus` is nonzero when `theta == 0` as well :)